tests/integration/benchmark/llmperf/scripts/run_benchmark.py

```diff
@@ -0,0 +1,514 @@
+
+        for line in output.stdout.splitlines():
+            logger.info(line)
+
+        for line in output.stderr.splitlines():
+            logger.error(line)
+
+        return output.stdout.strip()
+    except subprocess.CalledProcessError as e:
+        logger.error(f"Failed to run {bash_command}, error: {e}")
+        return None
+    except FileNotFoundError:
+        logger.error(f"Not found command: {bash_command}")
+        return None
+
+
+def pull_docker_image(container):
+    try:
+        image_name = get_image_name(container)
+        logger.info(f"image_name={image_name}")
+    except ValueError as e:
+        return None
+    bash_command = f"docker pull {image_name}"
+    if run_bash_command(bash_command.split()) is None:
+        return None
+    return image_name
+
+
+def build_parameter_str(parameter_list):
+    if len(parameter_list) == 0:
+        return ""
+    return " ".join(parameter_list)
+
+
+def launch_container(container, image, model, hf_token,
+                     container_log_file_name):
+    docker_parameters = build_parameter_str(
+        container.get("docker_parameters", []))
+    server_parameters = build_parameter_str(
+        container.get("server_parameters", []))
+
+    try:
+        container_name = container["container"]
+        docker_command = f"docker run --name {container_name} --rm -e HUGGING_FACE_HUB_TOKEN={hf_token} {docker_parameters} --init {image} {server_parameters}"
+        logger.info(f"docker_command={docker_command}")
+        # Open log file for writing
+        log_file = open(Path(container_log_file_name), "w")
+
+        process = subprocess.Popen(docker_command,
+                                   shell=True,
+                                   stdout=log_file,
+                                   stderr=subprocess.STDOUT)
+        logger.info(f"Docker run shell process={process.pid}")
+        time.sleep(60)
+        container = subprocess.check_output(
+            ["docker", "inspect", "-f", "{{.Name}}", container_name])
+    except Exception as e:
+        logger.error(f"Failed to launch container. error: {e}")
+        logger.error(traceback.format_exc())
+        raise e
+
+
+def wait_for_server(model, timeout=7200):
+    # wait for server to start return 1 if  server crashes
+    script = '''
+    #!/bin/bash
+    MODEL_ID=$1
+    TIMEOUT=$2
+    start_time=$(date +%s)
+    end_time=$((start_time + $TIMEOUT)) 
+    while [ $(date +%s) -lt $end_time ]; do
+        filter="$(curl -s http://0.0.0.0:8080/v1/models |jq -e --arg expected "$MODEL_ID" '. != null and . != {} and has("data") and .data != null and (.data | length > 0) and (.data[].id | contains($expected))')"
+        if [ -z "$filter" ]; then
+            echo "Model $MODEL_ID is not available"
+            sleep 1m
+        else
+            echo "Model $MODEL_ID is available"
+            exit 0
+        fi
+    done
+
+    echo "Model $MODEL_ID is not available within 2 hours"
+    exit 1
+    '''
```

Comment:
not a blocker, but I think this would easier to maintain in a standalone bash script rather than inline in the python script.

Reply:
actually, i see a few more inline bash scripts. I think some of them are pretty similar to other scripts we have, but would need to look into it. 

I'm fine with this for now - we can look into organizing some of these scripts post this pr